fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/ContainmentIndexImpl.java

```diff
@@ -144,91 +209,65 @@ private NamedParameterJdbcTemplate getNamedParameterJdbcTemplate() {
             // we are in a transaction
             parameterSource.addValue("transactionId", transactionId);
             final String currentChildrenQuery = SELECT_CHILDREN +
                     " UNION " + SELECT_ADDED_CHILDREN +
                     " EXCEPT " + SELECT_DELETED_CHILDREN;
-            for (final String child : jdbcTemplate.queryForList(currentChildrenQuery, parameterSource, String.class)) {
-                children.add(child);
-            }
+            children = jdbcTemplate.queryForList(currentChildrenQuery, parameterSource, String.class);
         } else {
             // not in a transaction
-            for (final String child : jdbcTemplate.queryForList(SELECT_CHILDREN, parameterSource, String.class)) {
-                children.add(child);
-            }
+            children = jdbcTemplate.queryForList(SELECT_CHILDREN, parameterSource, String.class);
         }
-        return children.build();
+        return children.stream();
     }
 
-    /**
-     * Close the database connection
-     */
-    @PreDestroy
-    private void cleanup() {
-        try {
-            conn.close();
-        } catch (SQLException e) {
-            e.printStackTrace();
-        }
-    }
-
-    /**
-     * Return a stream of fedora identifiers contained by the specified fedora resource.
-     *
-     * @param tx The transaction.  If no transaction, null is okay.
-     * @param fedoraResource The containing fedora resource
-     * @return A stream of contained identifiers
-     */
     @Override
     public Stream<String> getContainedBy(final Transaction tx, final FedoraResource fedoraResource) {
         final String txId = (tx != null) ? tx.getId() : null;
         return getChildren(fedoraResource.getId(), txId);
     }
 
-    /**
-     * Add a contained by relation between the child resource and its parent.
-     *
-     * @param tx The transaction.  If no transaction, null is okay.
-     * @param parent The containing fedora resource
-     * @param child The contained fedora resource
-     */
+    @Override
     public void addContainedBy(final Transaction tx, final FedoraResource parent, final FedoraResource child) {
-        final String txId = (tx != null) ? tx.getId() : null;
-        final NamedParameterJdbcTemplate jdbcTemplate = getNamedParameterJdbcTemplate();
+        final String txID = tx != null ? tx.getId() : null;
+        addContainedBy(txID, parent.getId(), child.getId());
+    }
+
+    @Override
+    public void addContainedBy(final String txID, final String parentID, final String childID) {
         final MapSqlParameterSource parameterSource = new MapSqlParameterSource();
-        parameterSource.addValue("parent", parent.getId());
-        parameterSource.addValue("child", child.getId());
-        if (txId != null) {
-            parameterSource.addValue("transactionId", txId);
-            jdbcTemplate.update(INSERT_CHILD_IN_TRANSACTION, parameterSource);
+        parameterSource.addValue("parent", parentID);
```

Comment:
There are multiple places throughout this class where literals are used ("child", "parent", transactionId). Should probably use the static constants defined at the top of the class.

Reply:
So technically the `child` here matches to the `:child` part of the SQL statement in.
```
"SELECT * FROM " + RESOURCES_TABLE + " WHERE " + FEDORA_ID_COLUMN + " = :child"
```

So we don't want to use the constants, but perhaps using a more descriptive name to differentiate between columns and arguments?

Reply:
To clarify, the above was a question. I'm not going to mess around with all the various SQL statements unless its required.

Reply:
I was not thinking of the SQL statements, but rather the String literals used in calls to `parameterSource.addValue()`. It seems more maintainable if "parent", "child" and "transactionId" were replaced by constants. 
Searching for "parameterSource.addValue(" shows 12 instances.

Reply:
To recap the Slack discussion, the replacing of literals with constants would work for the literals `parent` and `transactionId` due to them (fortuitously) being the same as the constants, however `child` would have to remain as a literal or be modified or have a new constant defined.

I'm not against this work, but I also don't see it as valuable and would prefer to push it off for now.

Reply:
Punting here is fine.